packages/commonwealth/server/workers/knock/eventHandlers/threadCreated.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  EventHandler,
+  logger,
+  notificationsProvider,
+  WorkflowKeys,
+} from '@hicommonwealth/core';
+import { models, Webhook } from '@hicommonwealth/model';
+import { safeTruncateBody } from '@hicommonwealth/shared';
+import { Op } from 'sequelize';
+import z from 'zod';
+import { config } from '../../../config';
+import { getProfileUrl, getThreadUrl } from '../util';
+
+const log = logger(import.meta);
+
+const output = z.boolean();
+
+export const processThreadCreated: EventHandler<
+  'ThreadCreated',
+  typeof output
+> = async ({ payload }) => {
+  const community = await models.Community.findOne({
+    where: { id: payload.community_id },
+  });
+  if (!community) {
+    log.error('Thread community not found!', undefined, {
+      payload,
+    });
+    return false;
+  }
+
+  const author = await models.Address.findOne({
+    where: { id: payload.address_id },
+    include: [{ model: models.User, required: true, attributes: ['profile'] }],
+  });
+
+  if (!author || !author.user_id) {
+    log.error('Full thread author with profile not found!', undefined, {
+      payload,
+    });
+    return false;
+  }
+
+  const webhooks = await models.Webhook.findAll({
+    where: {
+      community_id: community.id!,
+      events: { [Op.contains]: ['ThreadCreated'] },
+    },
+  });
+
+  if (webhooks.length > 0) {
+    const threadSummary = safeTruncateBody(
+      decodeURIComponent(payload.body!),
+      255,
+    );
+    const threadURl = getThreadUrl(
+      payload.community_id,
+      payload.id!,
+      community.custom_domain,
+    );
+
+    const thread = await models.Thread.findByPk(payload.id, {
+      attributes: ['title'],
+    });
+    const previewImg = Webhook.getPreviewImageUrl(
+      community,
+      decodeURIComponent(payload.body!),
+    );
+
+    const provider = notificationsProvider();
+
+    return await provider.triggerWorkflow({
+      key: WorkflowKeys.Webhooks,
+      users: webhooks.map((w) => ({
+        id: `webhook-${w.id}`,
+        webhook_url: w.url,
+        destination: w.destination,
+      })),
+      data: {
+        sender_username: 'Common',
+        sender_avatar_url: config.DEFAULT_COMMONWEALTH_LOGO,
+        community_id: community.id!,
+        title_prefix: 'New thread: ',
+        preview_image_url: previewImg.previewImageUrl,
+        preview_image_alt_text: previewImg.previewImageAltText,
+        profile_name:
+          author.User!.profile.name || author.address.substring(0, 8),
```

Comment:
won't this break if profile is undefined?

Reply:
`profile` is never undefined which is why this is not raising a type error. If the user hasn't set a profile then `User.profile` is set to an empty object so `profile.name` would resolve to `undefined`.